[Bug] compilation with ldb-toolchain failed when build thirdparty for openssl in ubuntu20.04

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

engines/e_afalg.c:40:11: fatal error: linux/if_alg.h: No such file or directory
   40 | # include <linux/if_alg.h>
      |           ^~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:6398: engines/e_afalg.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/cch13/Work/Project/incubator-doris/thirdparty/src/openssl-OpenSSL_1_1_1m'
make: *** [Makefile:174: all] Error 2

and we can see

~/Work/Project/incubator-doris$ ll /usr/include/linux/if_alg.h 
-rw-r--r-- 1 root root 1534 1月  12 23:16 /usr/include/linux/if_alg.h


### What You Expected?

build openssl succeed

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -265,9 +265,9 @@ build_thrift() {
     # NOTE(amos): libtool discard -static. --static works.
     ./configure CPPFLAGS="-I${TP_INCLUDE_DIR}" LDFLAGS="-L${TP_LIB_DIR} --static" LIBS="-lcrypto -ldl -lssl" CFLAGS="-fPIC" \
     --prefix=$TP_INSTALL_DIR --docdir=$TP_INSTALL_DIR/doc --enable-static --disable-shared --disable-tests \
-    --disable-tutorial --without-qt4 --without-qt5 --without-csharp --without-erlang --without-nodejs \
-    --without-lua --without-perl --without-php --without-php_extension --without-dart --without-ruby \
-    --without-haskell --without-go --without-haxe --without-d --without-python -without-java -without-rs --with-cpp \
+    --disable-tutorial --without-qt4 --without-qt5 --without-csharp --without-erlang --without-nodejs --without-nodets --without-swift \
+    --without-lua --without-perl --without-php --without-php_extension --without-dart --without-ruby --without-cl \
+    --without-haskell --without-go --without-haxe --without-d --without-python -without-java --without-dotnetcore -without-rs --with-cpp \
     --with-libevent=$TP_INSTALL_DIR --with-boost=$TP_INSTALL_DIR --with-openssl=$TP_INSTALL_DIR
 
     if [ -f compiler/cpp/thrifty.hh ];then

--- a/thirdparty/patches/mysql-server-mysql-5.7.18.patch
+++ b/thirdparty/patches/mysql-server-mysql-5.7.18.patch
@@ -51,3 +51,15 @@ diff -uprN a/cmd-line-utils/libedit/terminal.c b/cmd-line-utils/libedit/terminal
  
  
  	if (term == NULL)
+diff -uprN a/CMakeLists.txt b/CMakeLists.txt
+--- a/CMakeLists.txt    2022-01-23 23:30:48.915006943 +0800
++++ b/CMakeLists.txt    2022-01-23 23:31:04.785141731 +0800
+@@ -484,8 +484,6 @@ INCLUDE_DIRECTORIES(
+ MYSQL_CHECK_ZLIB_WITH_COMPRESS()
+ # Add bundled yassl/taocrypt or system openssl.
+ MYSQL_CHECK_SSL()
+-# Add system/bundled editline.
+-MYSQL_CHECK_EDITLINE()
+ # Add libevent
+ MYSQL_CHECK_LIBEVENT()
+ # Add lz4 library

